test/functional/unified-spec-runner/operations.ts

```diff
@@ -294,19 +324,36 @@ operations.set('insertMany', async ({ entities, operation }) => {
 });
 
 operations.set('iterateUntilDocumentOrError', async ({ entities, operation }) => {
-  const changeStream = entities.getEntity('stream', operation.object);
-  // Either change or error promise will finish
-  return Promise.race([
-    changeStream.eventCollector.waitAndShiftEvent('change'),
-    changeStream.eventCollector.waitAndShiftEvent('error')
-  ]);
+  try {
+    const changeStream = entities.getEntity('stream', operation.object);
+    // Either change or error promise will finish
+    return Promise.race([
+      changeStream.eventCollector.waitAndShiftEvent('change'),
+      changeStream.eventCollector.waitAndShiftEvent('error')
+    ]);
+  } catch (e) {
+    const findCursor = entities.getEntity('cursor', operation.object);
```

Comment:
Similar to the close operation I'm not understanding why this switches to cursor logic in the catch block.

Reply:
Same as above comment for close operation.